### PR TITLE
Increase decoupling between provisioning and installation

### PR DESCRIPTION
**Pull request by suleymanakbas91**
**Description**

Changes proposed in this pull request:

- Increase decoupling between provisioning and installation

**Related issue(s)**

See https://github.com/kyma-project/cli/issues/175
Also, solves https://github.com/kyma-project/cli/issues/168 and https://github.com/kyma-project/cli/issues/147

### CODE REVIEW

File: cmd/kyma/install/cmd.go [hunk 1]
```diff
@@ -893,3 +921,93 @@ func (cmd *command) importCertificate(ca trust.Certifier) error {
...
+		}
+	}
+
+	hostAlias = strings.Trim(IP, "\n") + hostnames
+
+	return addDevDomainsToEtcHostsOSSpecific(c.opts.Domain, s, hostAlias)
+}
+
+func (c *command) getClusterInfoFromConfigMap() (clusterInfo, error) {
+	var err error
```
Comment (igor-karpukhin) [inline cmd/kyma/install/cmd.go]: This statement can be removed

File: cmd/kyma/install/cmd.go [hunk 2]
```diff
@@ -175,6 +227,13 @@ func (cmd *command) validateFlags() error {
 			return fmt.Errorf("You specified 'installer-dir=%s' without specifying --local", cmd.opts.LocalInstallerDir)
 		}
 	}
+
+	// If one of the --domain, --tlsKey, or --tlsCert is specified, the others must be specified as well
+	if ((cmd.opts.Domain != "" && cmd.opts.Domain != localDomain) || cmd.opts.TLSKey != "" || cmd.opts.TLSCert != "") &&
+		!((cmd.opts.Domain != "" && cmd.opts.Domain != localDomain) && cmd.opts.TLSKey != "" && cmd.opts.TLSCert != "") {
+		return fmt.Errorf("You specified one of the --domain, --tlsKey, or --tlsCert without specifying the others. They must be specified together")
```
Comment (igor-karpukhin) [inline cmd/kyma/install/cmd.go]: Formatted output without arguments

File: cmd/kyma/install/cmd.go [hunk 3]
```diff
@@ -175,6 +227,13 @@ func (cmd *command) validateFlags() error {
 			return fmt.Errorf("You specified 'installer-dir=%s' without specifying --local", cmd.opts.LocalInstallerDir)
 		}
 	}
+
+	// If one of the --domain, --tlsKey, or --tlsCert is specified, the others must be specified as well
+	if ((cmd.opts.Domain != "" && cmd.opts.Domain != localDomain) || cmd.opts.TLSKey != "" || cmd.opts.TLSCert != "") &&
```
Comment (igor-karpukhin) [inline cmd/kyma/install/cmd.go]: This looks a bit complicated

**Comment by clebs** [comment]
As discussed offline and to keep track of the discussion:
We should aim to separate the gathering and compilation of installation resources (checking if local, from a release, from local sources, etc...) from the actual installation (deploying installer, watching, etc...).

- Keep the complexity of finding out **what** to install in one place without affecting the rest of the functionality.
- Installation is just a matter of deploying whatever resources are passed and manage the progress, etc...

**Pull request merged**